### PR TITLE
add shape_check for op tests

### PR DESCRIPTION
**Pull request by zhupengyang**
- add check for op input shape in Op unittests
- op input shape must be greater than or equal 100
- add need_to_fix_check_shape_op_list to avoid existing ops
- check incremental ops

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/op_test.py [hunk 1]
```diff
@@ -164,6 +170,13 @@ def tearDownClass(cls):
 
         _set_use_system_allocator(cls._use_system_allocator)
 
+        if hasattr(
+                get_numeric_gradient, 'check_shape_time'
+        ) and get_numeric_gradient.check_shape_time == 0 and OpTest.op_type not in need_to_fix_check_shape_op_list:
```
Comment (hong19860320) [inline python/paddle/fluid/tests/unittests/op_test.py]: 只要保证单测输入tensor中有一个tensor的size大于100就算pass吗？

Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/op_test.py]: 是的

File: python/paddle/fluid/tests/unittests/check_shape_whitelist.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    'sequence_softmax', 'reduce_sum', 'nce', 'sigmoid_focal_loss', 'sum',
+    'elementwise_pow', 'pad_constant_like', 'squeeze2', 'norm', 'unsqueeze',
+    'fused_elemwise_activation', 'tree_conv', 'huber_loss', 'group_norm',
+    'bilinear_tensor_product', 'flatten2', 'unsqueeze2', 'mul', 'im2sequence',
+    'kldiv_loss', 'unpool', 'cos_sim', 'strided_slice', 'cast', 'reverse',
+    'hinge_loss', 'flatten', 'scatter_nd_add', 'expand_as',
+    'teacher_student_sigmoid_loss', 'label_smooth', 'spectral_norm',
+    'elementwise_min', 'abs', 'acos', 'warpctc', 'nearest_interp', 'data_norm',
+    'match_matrix_tensor', 'var_conv_2d', 'fused_embedding_seq_pool',
+    'increment', 'elementwise_mul'
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/check_shape_whitelist.py]: 有些OP可以通过简单改一下单测的尺寸用例，就可以过了。这些并不是真正有问题的Op名单。

Comment (luotao1) [reply python/paddle/fluid/tests/unittests/check_shape_whitelist.py]: 为了便于管理，请放到`python/paddle/fluid/tests/unittests/white_list`目录下。

Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/check_shape_whitelist.py]: Done

File: python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+NOT_CHECK_OP_LIST = [
+    # ngraph: increment's input must be 1-d and only has one data
+    'increment'
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: @baojun-nervana @LeoZhao-Intel Please review it

Comment (baojun-nervana) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: Does ngraph increase op cause any problem here? You may disable increase op by comment out this line https://github.com/NervanaSystems/ngraph-paddle/blob/develop/paddle/fluid/operators/ngraph/ops/increment_op.h#L49. We then can submit following PR to fix the op.

Comment (LeoZhao-BJ) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: can author provide the use case or expected behavior for increase op? not very clear if just judge from comment here:

# ngraph: increment's input must be 1-d and only has one data

Comment (luotao1) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: @LeoZhao-Intel We want each Op to contain a unit test with dimensions of input >=100.
@baojun-nervana You can remove `increment` here, and you may find the problem.

Comment (LeoZhao-BJ) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: how to run this  unittest to reproduce? can yo share cmd?

Comment (luotao1) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: `ctest -R test_increment_ngraph_op -V`

Comment (LeoZhao-BJ) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: seems increment op has restriction on input dim, must be 1.

https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/increment_op.cc#L34

test case may be invalid.  so in this case, putting it into NOT_CHECK_OP_LIST is correct.

Comment (LeoZhao-BJ) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api_cn/layers_cn/increment_cn.html

**Comment by LeoZhao-Intel** [comment]
LGTM

File: python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py (hunk 3 above)
Comment (luotao1) [reply python/paddle/fluid/tests/unittests/white_list/check_shape_white_list.py:17]: Got it. Thanks @LeoZhao-Intel 
> ngraph: increment's input must be 1-d and only has one data

@zhupengyang the notes is not correct, it is not just for ngraph. see https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api_cn/layers_cn/increment_cn.html

**Pull request merged**